File: store-backend/src/main/java/com/springboot/everefing/service/OrderService.java
package com.springboot.everefing.service;

import com.springboot.everefing.dto.OrderDto;
import com.springboot.everefing.entity.Order;

import java.util.List;
import java.util.Optional;

public interface OrderService {
    OrderDto addOrder(OrderDto orderDto);
    OrderDto getOrder(Long orderId);
    List<OrderDto> getAllOrders();
    void deleteOrder(Long orderId);

    OrderDto increaseQuantity(Long orderId);
    OrderDto reduceQuantity(Long orderId);
    List<OrderDto> findByUsername(String username);
    void deleteByUsername(String username);
}

File: store-front/src/services/OrderService.js
import axios from "axios";

const BASE_ORDER_REST_API_URL = 'http://localhost:8080/api/orders'

export const addOrder = (order) => axios.post(BASE_ORDER_REST_API_URL, order)

export const findOrderByUsername = (username) => axios.get(BASE_ORDER_REST_API_URL + '/' + username + '/findByUsername')

export const deleteByUsername = (username) => axios.delete(BASE_ORDER_REST_API_URL + '/'+ username + '/deleteByUsername')

export const increaseQuantity = (id) => axios.patch(BASE_ORDER_REST_API_URL + '/' + id + '/increase')
export const reduceQuantity = (id) => axios.patch(BASE_ORDER_REST_API_URL + '/' + id + '/reduce')

export const deleteOrder = (id) => axios.delete(BASE_ORDER_REST_API_URL + '/' + id)